File: photomap-core-api/photo-api/src/main/java/ua/in/photomap/photoapi/mapper/LicenseMapper.java
package ua.in.photomap.photoapi.mapper;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.Mappings;
import org.mapstruct.factory.Mappers;
import ua.in.photomap.photoapi.dto.LicenseDTO;
import ua.in.photomap.photoapi.model.License;

@Mapper
public interface LicenseMapper {
    LicenseMapper INSTANCE = Mappers.getMapper(LicenseMapper.class);

    @Mappings(
            {
                    @Mapping(target = "id", source = "id"),
                    @Mapping(target = "name", source = "name"),
                    @Mapping(target = "description", source = "description"),
                    @Mapping(target = "detailsUrl", source = "detailsUrl")
            }
    )
    LicenseDTO licenseToLicenseDto(License entity);
}

File: photomap-core-api/photo-api-python/mappers/license_mapper.py
from models.license import License
from dto.license_dto import LicenseDTO

class LicenseMapper:
    def license_to_license_dto(self, license: License) -> LicenseDTO:
        return LicenseDTO(
            id=license.id,
            name=license.name,
            description=license.description,
            detailsUrl=license.details_url
        ) 